packages/next/src/server/base-server.ts

```diff
@@ -1741,6 +1746,45 @@ export default abstract class Server<ServerOptions extends Options = Options> {
     }
   }
 
+  protected pathCouldBeIntercepted(resolvedPathname: string): boolean {
+    return (
+      isInterceptionRouteAppPath(resolvedPathname) ||
+      this.interceptionRouteRewrites?.some((rewrite) => {
+        return new RegExp(rewrite.regex).test(resolvedPathname)
+      })
+    )
+  }
+
+  protected setVaryHeader(
+    req: BaseNextRequest,
+    res: BaseNextResponse,
+    isAppPath: boolean,
+    resolvedPathname: string
+  ): void {
+    const baseVaryHeader = `${RSC_HEADER}, ${NEXT_ROUTER_STATE_TREE}, ${NEXT_ROUTER_PREFETCH_HEADER}`
+    const isRSCRequest =
+      Boolean(req.headers[RSC_HEADER.toLowerCase()]) ||
```

Comment:
Rather than using `Boolean(...)`, we know that the value should be `1`, so why not do:

```suggestion
      req.headers[RSC_HEADER.toLowerCase()] === '1' ||
```